File: unique-paths/unique-paths.js
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function(m, n, memo = new Map()) {
    const key = `${m}#${n}`
    if (memo.has(key)) return memo.get(key)
    
    if (m === 1 && n === 1) return 1
    if (m < 0 || n < 0) return 0
    
    memo.set(key, uniquePaths(m - 1, n, memo) + uniquePaths(m, n - 1, memo))
    return memo.get(key) 
};

File: unique-paths/unique-paths.go
func uniquePaths(m int, n int) int {
    dp := make([][]int, m)
    
    for i := 0; i < m; i++ {
        dp[i] = make([]int, n)
    }
    
    dp[0][0] = 1
    
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ { 
            current := dp[i][j]
            if j + 1 < n {
                dp[i][j + 1] += current
            }
            if i + 1 < m {
                dp[i + 1][j] += current
            }
        }
    }
    
    
    
    return dp[m - 1][n - 1]
}